src/Behat/Behat/Output/Node/EventListener/JUnit/JUnitOutlineStoreListener.php

```diff
@@ -53,7 +54,18 @@ public function __construct(SuitePrinter $suitePrinter)
      */
     public function listenEvent(Formatter $formatter, Event $event, $eventName)
     {
-        $this->captureOutlineOnBeforeOutlineEvent($event);
+      $this->captureOutlineOnBeforeOutlineEvent($event);
+      if (method_exists($event, 'getSpecificationIterator')) {
```

Comment:
Instead of this method check, I would check if the event is an instance of the [`SuiteTested` class](https://github.com/Behat/Behat/blob/fb3241979bd080ac17f157244e9d6fc84a22bf9d/src/Behat/Testwork/EventDispatcher/Event/SuiteTested.php), that way you are sure that the return value is the expected type.

Even better (but not mandatory here I would say - @everzet to confirm) would be to introduce an interface for the events that contains this method, such as the [hook `SuiteScope` interface](https://github.com/Behat/Behat/blob/fb3241979bd080ac17f157244e9d6fc84a22bf9d/src/Behat/Testwork/Hook/Scope/SuiteScope.php).